tests/test_daemon.cpp

```diff
@@ -1336,9 +1341,9 @@ TEST_F(Daemon, writesAndReadsMountsInJson)
 
     // Create a temp folder containing three subfolders, and create mount points for all of them.
     auto temp_mount_dir = QDir(mpt::TempDir().path());
-    auto temp_mount_1 = make_dir(temp_mount_dir, QString("a"), QFileDevice::Permissions{}).toStdString();
-    auto temp_mount_2 = make_dir(temp_mount_dir, QString("b"), QFileDevice::Permissions{}).toStdString();
-    auto temp_mount_3 = make_dir(temp_mount_dir, QString("c"), QFileDevice::Permissions{}).toStdString();
+    auto temp_mount_1 = MP_UTILS.make_dir(temp_mount_dir, QString("a"), QFileDevice::Permissions{}).toStdString();
+    auto temp_mount_2 = MP_UTILS.make_dir(temp_mount_dir, QString("b"), QFileDevice::Permissions{}).toStdString();
+    auto temp_mount_3 = MP_UTILS.make_dir(temp_mount_dir, QString("c"), QFileDevice::Permissions{}).toStdString();
```

Comment:
The permissions here are empty, but there is no special treatment for empty permissions anymore. I am not sure how this worked. Perhaps the platform mock was not doing anything to set the permissions. In the alternative above, I just bypass all that with a dedicated lambda, but feel free to use your trick to call the base here, as long as we're sure the permissions are handled correctly.

Reply:
Maybe each temp dir could be just `QDir(mpt::TempDir().path());`?

Reply:
`QDir(mpt::TempDir().path())` doesn't work since it doesn't actually create a directory.

Reply:
Doesn't `mpt::TempDir()` create a temporary directory?

Reply:
I found that it wasn't creating anything. And then the test would fail.

Reply:
Oh right, because that `TempDir` is a temporary in that expression, so it would delete the directory after creating the `QDir`. You'd have to bind it to a variable. Anyway, it's fine this way too.